src/libYARP_OS/src/PortReaderBufferBase.cpp

```diff
@@ -572,35 +572,24 @@ bool PortReaderBufferBase::acceptObjectBase(PortReader *obj,
             mPriv->consumeSema.wait();
         }
     }
-    bool ok = true;
-    if (ok) {
-        reader->setExternal(obj, wrapper);
 
```

Comment:
The reason for this is a few lines above:

```
/// Careful!  merge with ::read -- very similar code
/// Until merge, don't change one without looking at other :-(
```

I'm skipping this one for now, let's have a better look at the code before merging this